doc/whats_new/v1.4.rst

```diff
@@ -44,6 +44,11 @@ Changes impacting all modules
 
   :pr:`28235` by :user:`Julien Jerphanion <jjerphan>`.
 
+- |Fix| The adapters handling the creation of DataFrame now rename properly the
+  column names by handling duplicates. It notably happens when using the
+  :class:`~compose.ColumnTransformer`.
```

Comment:
I think we need 2 entries.

The one for all modules it is due of using a list instead of a dict and impact all transformers when using `set_output`.

The second bug is related to `ColumnTransformer` only requesting polars where the new logic allow to get a better error message and allow to transform in some cases.